client/src/globals/UserContext.js

```diff
@@ -1,13 +1,29 @@
 import React, { useState, createContext, useEffect } from "react";
+import CircularProgress from '@material-ui/core/CircularProgress';
+import { makeStyles } from '@material-ui/core/styles';
+import Grid from '@material-ui/core/Grid';
+import { Route } from 'react-router-dom';
+
+const useStyles = makeStyles((theme) => ({
+    loadingContainer: {
+        paddingTop: 300
+    }
+}))
 
 export const UserContext = createContext({
     userId: "",
-    setUserId: () => {}
+    setUserId: () => {},
+    isSubscribed:false,
+    setIsSubscribed: ()=>{}
   });
 
 export const UserContextProvider = (props) => {
     const [userId, setUserId] = useState("");
-    const value = { userId, setUserId };
+    const [isSubscribed, setIsSubscribed] = useState(false)
```

Comment:
user email is required to check whether user subscription is valid or not on the backend, so i added in user context. Alot of frontend pages also require user email, but are still fetching from backend. If we end up having extra time, we can refactor those code.

Reply:
Great Work :+1: